### PR TITLE
Implement three javalib CollectorsTests for JDK > 8

### PR DESCRIPTION
**Pull request by LeeTibbert**
* Implement CollectorsTests for methods introduced in JDK9, JKD10, & JDK12.
   This implementation requires Scala 3 because it uses lambda to provide a
   better model of idiomatic use.

* Remove no longer needed BuiltTestOnJDx scaffolding files.  They have served their purpose.

Tested that the tests ran where they should and did not run where the should not by
increasing build logging and attempting to run the tests. Visual inspection of the output
log would show if the tests ran or not.  All behaved as expected.

### CODE REVIEW

**Comment by ekrich** [comment]
Are you saying you have used special Scala 3 sauce in the implementation so that these new Java classes can only be used with Scala 3? If so can you show an example?

Is this a not so subtle way to get people to upgrade? :-)

**Review by WojciechMazur — APPROVED** [review]

**Comment by LeeTibbert** [comment]
@ekrich Thank you for motivation me to expand on the Scala3 requirement
and capture it for future readers.

The Collectors, the units-under-test (UUTs), and javalib stream methods in
general should work on all supported Scala versions, given the expected
JDK version [^1].

TL; DR; The secret sauce is in the Scala 3 compiler, not in any SN javalib code.

The Scala 3 restriction comes from the ease, or lack thereof, of the setup
needed to call the routine.  For the more complex Collectors, the cost
of setup on Scala 2.13 and, especially  2.12, is likely to outweigh the benefit of the call.

Scala 3 allows an idiomatic call such as:
```
val collector0 =
      Collectors.teeing(
        Collectors.counting(),
        Collectors.filtering(
          (e: Employee) => e.department == "LKG",
          Collectors.counting()
        ),
        (r1, r2) => Arrays.asList(r1, r2)
      )
```

Here, one has to specify the type of element in one of the lambdas but otherwise the code is pretty
straight forward, if dense,

Scala 2 has a harder time figuring out the rather complex type parameters involved.  It tends to 
easier to work with by using separate and explicit `Function`, `BiFunction`, etc as arguments
rather than lambdas. 

Separate Scala 2.13 tests could be written (I know, I have done many of them whilst verifying PR #3268 ) ,
but it is hard to get the resultant code to be presentation (PR) quality.  It is questionable if anyone
would use the more complex Collectors with Scala 2.13.  

To my thinking, it is a judgement call as to the most effective way to spend development time.
If the lack of 2.13 use models becomes an issue, I can revisit publishing those tests.

Scala 2.12 tests are an exercise in obfuscated Scala. I doubt anyone would use the more complex 
Collectors with Scala 2.12, even in ported code. The changes would be too extensive.

Does this description make sense?

[^1]: `Stream.of(singleton)` on Scala 2.12 requires special calling handling, shown in
          tests and description of PR #3268.

**Comment by ekrich** [comment]
Thank you for this great explanation. I think this is a totally reasonable design based on the circumstances. If this encourages someone to use Scala 3 for this capability then I am all for that choice. Plus if these new APIs encourage people from Java land to switch, all the better.

**Comment by LeeTibbert** [comment]
I appreciate the feedback, this is indeed a Community effort.
